tests/integration/test_kubernetes_control_plane_integration.py

```diff
@@ -164,8 +165,13 @@ async def _auth_req(token, timeout=30):
     assert not any(await asyncio.gather(*tasks))
 
 
-async def test_pod_security_policy(ops_test, kubernetes):
+async def test_pod_security_policy(pytestconfig, ops_test, kubernetes):
     """Test the pod-security-policy config option"""
+    channel = pytestconfig.getoption("--snap-channel")
+    track, risk = channel.split("/", 1)
+    if float(track) >= 1.25:
```

Comment:
If we cast versions to floats for comparisons, the comparisons will not always make sense. For example, `1.9` is greater than `1.25`, and `1.100` is less than `1.25`.

I think this particular case is unlikely to cause us problems for a long long time, but I think it's good practice to avoid casting versions to floats. I've had good results doing comparisons on tuples of integers instead:

```
if tuple(int(x) for x in track.split('.')) >= (1, 25):
```